core/src/main/java/org/infinispan/distribution/ch/impl/HashFunctionPartitioner.java

```diff
@@ -60,12 +63,7 @@ public boolean equals(Object o) {
          return true;
       if (o == null || getClass() != o.getClass())
          return false;
-
-      HashFunctionPartitioner that = (HashFunctionPartitioner) o;
-
-      if (numSegments != that.numSegments)
-         return false;
-      return Objects.equals(hashFunction, that.hashFunction);
+      return true;
```

Comment:
So all `HashFunctionPartitioner`s are equal to each other? What about eh 3 different variables? Also the `hashCode` is broken as that uses 2 of the fields :(

Reply:
@tristantarrant ^

Reply:
I guess we should consider all fields of HashFunctionPartitioner as transient, since those are set at runtime from other parts of the configuration.

Reply:
Ok, I've solved with https://github.com/infinispan/infinispan/pull/5556/files#diff-f50db7fa1d8dc90862597d5393c6795bR64 and restored the equality.
Bear in mind that segmentSize is a field computed from numSegments, so doesn't affect equality or hashcode